### ISSUE TITLE
[BUG] ObjectCache continually grows as project is played and paused

### ISSUE THREAD

**Issue by Luxapodular** [Wicklets/wick-editor#123]
**Describe the bug**
The object cache is not properly caching object references. 

**Expected behavior**
The object cache should have the same number of objects between play and pause.

**Do you have a suggested solution to this issue? (ex. has another program fixed this bug a certain way? Are you familiar with where in the code base someone would need to fix this issue?)**

It looks like the object UUID's are changing between plays, which causes new objects to be cached.

**Comment by Luxapodular** [comment]
I've started to take a look into Wick.ObjectCache.js. It looks like our initial thoughts (UUIDs) are changing, was wrong! UUIDs seem to be staying the same. Instead, it looks like copies of the existing Wick Objects on screen are being created. 

Things we know:
- This happens on Play/Pause
- Copies do not exponentially increase. (The copies occur only of real objects that are in the project)

Every Play Pause, the object cache receives unnecessary copies of the objects in the project, in addition to the actual objects themselves. These seem to be actual COPIES, with new UUIDs.

On Play/Pause an empty project receives:
- 2 copies of the Selection
- 3 copies of each Frame
- 3 copies of each Layer
- 3 copies of each Timeline
- 2 copies of each Clip
- 1 copy of the Project

Paths do not seem to be copied.

Additionally:
- Clips, Buttons, Frames, and Layers that are added after the first set of objects seem to only be copied ONCE per play/pause. This makes me hopeful that there is a simple, single bug causing most of the copying issues.

Strange bug, but I'm sure we'll figure this one out soon!

**Comment by Luxapodular** [comment]
A bit more context: It looks like there are only 5 places in the engine where ObjectCache.addObject is called. 3 Times in Base.js and 2 Times in WickFile.js. 

The copies are likely occurring when the "quick saved" project is deserialized.

These lines in WickFile.js are the culprits for the "Single" copies of elements. These calls are unecessary when play/pausing because we've already cached the objects anyway.

```
                for(var uuid in projectData.objects) {
                    var data = projectData.objects[uuid];
                    var object = Wick.Base.fromData(data);
                    Wick.ObjectCache.addObject(object); // These objects are already in the object cache between play/pauses.
                }
                var project = Wick.Base.fromData(projectData.project);
                Wick.ObjectCache.addObject(project); // This is where the extra copy of the project is being added to the object cache.
```

fromData() is the other culprit. These problems will likely be mitigated when we upgrade the autosave system but this is a need to discuss with @zrispo.

**Comment by Luxapodular** [comment]
Added a few tests to catch these issues in the fix-objectCache branch under commit 4d7feb06470c801cdbdd360727c52507bd59590c
